File: implementations/rust/src/years/y2022/day_4/part_1.rs
use super::common::{Pair, parse_input};

pub fn execute(input: String) -> String {
    return parse_input(&input)
        .filter(|Pair { x, y }| x.fully_contains(y) || y.fully_contains(x))
        .count()
        .to_string();
}

File: implementations/node/core/years/2022/day-4/part-1.mjs
import { parseInput } from "./common.mjs";

function fullyContains(pair1, pair2) {
    return pair1[0] <= pair2[0] && pair1[1] >= pair2[1]
}

export function execute(input) {
    return parseInput(input)
        .filter(([pair1, pair2]) => fullyContains(pair1, pair2) || fullyContains(pair2, pair1))
        .length
}
